### PR TITLE
Adding the phone number endpoint so that you can check which numbers …

### PR DESCRIPTION
**Pull request by nbragdon**
This adds the basics of the phone number endpoint. This allows you to request phone number information using the sdk. This also allows you to allocate new numbers. This implementation was a fairly light weight wrapper though. Feel free to give suggestions if this isn't the direction you wanted it to go in.

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -45,9 +45,9 @@ var Client = function (config) {
 		return response;
 	}
 
-	function getUserAgentHeader() {
-		return packageInfo.name + "-v" + packageInfo.version;
-	}
+	this.makeRequest = function (params) {
```
Comment (dtolb) [inline lib/client.js:48]: is this just switching these around for style purposes?

File: lib/phoneNumber.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+var PhoneNumber = function (client) {
+	/**
+		 * Gets a list of all your phone numbers.
```
Comment (dtolb) [inline lib/phoneNumber.js:3]: Fix formatting in comments...

File: test/message-test.js [hunk 3]
```diff
@@ -167,5 +174,29 @@ describe("Message API", function () {
...
+		it("should get try to get the next page of results, promise style", function (done) {
+			client.Message.list({
+				fromDateTime : fromDateTime,
+				toDateTime   : toDateTime
+			})
+			.then(function (messageResponse) {
+				return messageResponse.getNextPage();
+			})
+      .then(function (messageResponse) {
+	var messages = messageResponse.messages;
```
Comment (dtolb) [inline test/message-test.js:186]: formatting :(

File: test/number-test.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	describe("global methods", function () {
+		var client;
+
+		var userId = "fakeUserId";
+		var apiToken = "fakeApiToken";
+		var apiSecret = "fakeapiSecret";
+
+		var phoneNumberList = [
+      {
+	"id"             : "{numberId1}",
```
Comment (dtolb) [inline test/number-test.js:17]: formatting

File: test/number-test.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+				numberResponse[1].numberState.should.equal(phoneNumberList[1].numberState);
+
+				done();
+			});
+		});
+
+		it("phone number should be allocated", function (done) {
+			client.PhoneNumber.allocate()
+      .then(function (response) {
+	done();
```
Comment (dtolb) [inline test/number-test.js]: formatting

File: test/number-test.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+		});
+
+		after(function () {
+			nock.cleanAll();
+			nock.enableNetConnect();
+		});
+
+		it("should get a list of phone numbers, promise style", function (done) {
+			client.PhoneNumber.list()
+			.then(function (numberResponse) {
```
Comment (dtolb) [inline test/number-test.js:67]: prefer to use deep equal here 

or break this down into individual `it` statements. Ideally, only one assertion per `it` block if possible. Helps trace down future errors.

File: test/number-test.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+			.done(done);
+		});
+
+		it("should get a list of phone numbers, callback style", function (done) {
+			client.PhoneNumber.list(null, function (err, numberResponse) {
+				if (err) {
+					throw err;
+				}
+
+				numberResponse[0].id.should.equal(phoneNumberList[0].id);
```
Comment (dtolb) [inline test/number-test.js:97]: Same here as above

File: test/number-test.js [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+				numberResponse[1].createdTime.should.equal(phoneNumberList[1].createdTime);
+				numberResponse[1].state.should.equal(phoneNumberList[1].state);
+				numberResponse[1].price.should.equal(phoneNumberList[1].price);
+				numberResponse[1].numberState.should.equal(phoneNumberList[1].numberState);
+
+				done();
+			});
+		});
+
+		it("phone number should be allocated", function (done) {
```
Comment (dtolb) [inline test/number-test.js:121]: What is this testing? Missing assertion?

File: test/number-test.js (hunk 5 above)
Comment (dtolb) [inline test/number-test.js:124]: formatting

File: test/number-test.js [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+				apiSecret : apiSecret
+			});
+			nock.disableNetConnect();
+
+			nock(baseUrl)
+				.persist()
+				.post("/v1/users/" + userId + "/phoneNumbers")
+				.reply(201, {},
+          {
+	"Location" : "/v1/users/" + userId + "/phoneNumbers/fakeNumberId"
```
Comment (dtolb) [inline test/number-test.js:53]: formatting

File: test/number-test.js [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+	"number"         :"{number1}",
+	"nationalNumber" :"{national_number1}",
+	"name"           : "home phone",
+	"createdTime"    : "2013-02-13T17:46:08.374Z",
+	"state"          : "NC",
+	"price"          : "0.60",
+	"numberState"    : "enabled"
+      },
+      {
+	"id"             : "{numberId2}",
```
Comment (dtolb) [inline test/number-test.js:28]: formatting

File: test/message-test.js [hunk 11]
```diff
@@ -167,5 +174,29 @@ describe("Message API", function () {
 			});
 		});
 
+		it("should get try to get the next page of results, promise style", function (done) {
```
Comment (dtolb) [inline test/message-test.js:177]: Prefer to have just one assertion per `it` block. 

Roll in deep equals?

**Comment by dtolb** [comment]
@nbragdon Looks like some formatting grabbed you again somewhere? What editor are you using, let's see if we can figure out what's happening there.

For `it` statements, prefer to keep to one assertion per.  Check out [Deep Equals](https://shouldjs.github.io/#should-deepequal) when comparing JSON.

File: lib/client.js (hunk 1 above)
Comment (nbragdon) [inline lib/client.js:48]: Yeah I didn't even actually make those changes. It was automatically done when I used the command to fix styling.

File: lib/phoneNumber.js [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+		 * @param {Object} params Query parameters for sorting the results
+		 * @param {String} [params.page=0] Used for pagination to indicate the page
```
Comment (fuchsnj) [inline lib/phoneNumber.js:5]: Please don't document that page parameter. This is deprecated and should not be used. the 'next' header link should be used instead

File: lib/phoneNumber.js [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+		 * @param {Number} [params.name] Used to filter the retrieved list of numbers
+	   *  allocated for the authenticated user by it's name.
+		 * @param {Number} [params.city] Used to filter the retrieved list of numbers
+	   *  allocated for the authenticated user by it's city.
+	   * @param {Number} [params.numberState] Used to filter the retrieved
+	   *  list of numbers allocated for the authenticated user by the number state.
+		 * @param {Function} callback A callback with the list of numbers
+		 * @return {Promise} A promise for the list of numbers
+		 */
+	this.list = function (params, callback) {
```
Comment (fuchsnj) [inline lib/phoneNumber.js]: Why is `list` an instance of PhoneNumber? I would have to create a PhoneNumber object just to just my PhoneNumbers?

**Comment by dtolb** [comment]
@nbragdon need to merge master back into here :(

**Comment by dtolb** [comment]
@nbragdon any plans to update this? Or should I just take this over

**Comment by dtolb** [comment]
Closing

**Pull request closed without merging**